src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int capacity = DEFAULT_CAPACITY;
+    private T[] storage;
+    private int current = 0;
+
+    public ArrayList() {
+        storage = (T[]) new Object[capacity];
+    }
+
+    private T[] resize() {
+        capacity = (capacity * 3) / 2 + 1;
+        T[] newArray = (T[]) new Object[capacity];
+        System.arraycopy(storage, 0, newArray, 0, current);
+        return newArray;
+    }
 
     @Override
     public void add(T value) {
-
+        if (capacity == current) {
+            storage = resize();
+        }
+        storage[current++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        while (index > capacity) {
```

Comment:
if in any method you receive an index that is larger than your current `storage.length`, you should throw an Exception